src/Check/DoctrineMigration.php

```diff
@@ -35,19 +65,35 @@ public function __construct(Configuration $migrationConfiguration)
      */
     public function check()
     {
-        $availableVersions = $this->migrationConfiguration->getAvailableVersions();
-        $migratedVersions = $this->migrationConfiguration->getMigratedVersions();
-
-        $notMigratedVersions = array_diff($availableVersions, $migratedVersions);
+        $notMigratedVersions = array_diff($this->availableVersions, $this->migratedVersions);
         if (! empty($notMigratedVersions)) {
             return new Failure('Not all migrations applied', $notMigratedVersions);
         }
 
-        $notAvailableVersion = array_diff($migratedVersions, $availableVersions);
+        $notAvailableVersion = array_diff($this->migratedVersions, $this->availableVersions);
         if (! empty($notAvailableVersion)) {
             return new Failure('Migrations applied which are not available', $notMigratedVersions);
         }
 
         return new Success();
     }
+
+
+    private function getAvailableVersionsFromDependencyFactory(DependencyFactory $dependencyFactory)
```

Comment:
The return type for this should be documented and well-typed (it is an `array`: let's document the element types too)